### ISSUE TITLE
"Toggle Attack Detections" doesn't remain persistent after re-start

### ISSUE THREAD

**Issue by E3V3A** [SecUpwN/Android-IMSI-Catcher-Detector#578]
If you try to disable "Toggle Attack Detections", your choice is not remembered after restart/quitting of app. I suppose we need to save this as a setting somewhere.

**Comment by SecUpwN** [comment]
Confirming this. Added to this Issue it might be a good idea to save all settings in `sdcard/Android/data/com.SecUpwN.AIMSICD/files/settings.xxx` to enable users to use their old settings after a fresh install.

**Comment by DJaeger** [comment]
We have a setting in the preferences activity to toggle the autostart of this feature.
And it works!?
What do you mean?

**Comment by DJaeger** [comment]
I would suggest to close this, as all is working as expected

**Comment by SecUpwN** [comment]
>I would suggest to close this, as all is working as expected

Agreed. @E3V3A, feel free to re-open this with an extended explanation, should this happen again.

**Comment by E3V3A** [comment]
I see, the problem is the different names in Preferences, versus on Drawer:

On **Drawer**:
```
Toggle Attack Detection
Toggle Cell Tracking
```
In **Preferences**:
```
Cell Tracking
Cell Monitoring
```

At this point I have no ****ing idea what is what... And if they are the same or different.

**Comment by DJaeger** [comment]
They are for the same but simply differently named.
I also had to test which one in the preferences is for which one in the navigation drawer.
In addition they are in reversed order

**Comment by SecUpwN** [comment]
>They are for the same but simply differently named.
>In addition they are in reversed order.

I can totally understand @E3V3A's frustration here and am sure users will appreciate a simplification. Maybe removing the stuff from the `Preferences` and just keeping it in `Navigation Drawer` will help a bit?

**Comment by E3V3A** [comment]
> Maybe removing the stuff from the `Preferences` and just keeping it in `Navigation Drawer`

I agree, but keep it as a remembered "preference" when toggling.

**Comment by DJaeger** [comment]
We shouldn't have some feature switches in preferences and some in the navigation drawer.
I think we should move them all to preferences and let them all act immediately and not only after restart, like it is done already for silent SMS and femtocell detection.
This would also make the navigation drawer shorter.

**Comment by E3V3A** [comment]
Yes and no. You do want a quick toggle for when you're moving around or going inside etc. But at the same time the data will be much less useful if users keep on toggling back and forth without knowing/understanding why...

**Comment by DJaeger** [comment]
I think it should never be toggled "quick".
You should use the feature or not, as you can be a target of MITM everywhere, *especially* at home!

**Comment by E3V3A** [comment]
Well, one of the toggles is about collecting the data and requiring GPS...

**Comment by ga900** [comment]
If may I suggest to use gps fix flag to turn on-off data collecting? If gps has fix collect data otherwise stop collecting. For detection use position from last fix. Of course option to turn on and off mentioned options in some advanced setup would be nice :).

**Comment by DJaeger** [comment]
@ga900: we already have an issue for this somewhere

**Comment by TPS** [comment]
@DJaeger @ga900 Do you both mean #251?

**Comment by DJaeger** [comment]
@TPS: Yes, thanks.

**Comment by E3V3A** [comment]
@ga900 
> ... use gps fix flag to turn on-off data collecting? If gps has fix collect data otherwise stop collecting. For detection use position from last fix.

Well, this is where things get a bit hairy...
Because, blocking GPS is very easy, we made #309. In addition, we still wanna keep an eye on the connected network info, so please clarify what you mean should happen. 

So perhaps:
 1. One toggle should be to log GPS positions or not.
 2. The other to log (in file/db) other events, but not disable warnings about type-0 SMS, changing LAC etc. but only toggle the .

**Initially** the idea was that one of the buttons should lock the phone into 2G-mode only, for detection. But since then, it seem that LTE provides a better security and the TA, PD parameters needed to better detect an IMSIC, so perhaps the second toggle should be updated to be, kind of, `Lock phone to LTE-only`?

**Comment by SecUpwN** [comment]
Erm.. may I drop my own suggestion here? How about this:

* Rename the toggle buttons and buttons in `Preferences` so that users know they're the same
* Sync the buttons  - when toggling via Navigation Drawer, the button in `Preferences` changes state

**Comment by DJaeger** [comment]
@SecUpwN: But why have them on two locations?
And if they are at two locations why not also the others (silent sms, femtocell...) ?

**Comment by E3V3A** [comment]
That is a good temp solution, but it would be redundant and still confusing with two locations.

**Comment by ga900** [comment]
If you lose GPS you probably entered building, home, underground garage, shopping mall or tunnel. Except in longer tunnels, where is expected to have different LACs on entry and exit, in all other places I would expect to stay in the same LAC, so I would use last known position for detection of threats and measurements, but will not upload such measurements to OCID. So I would put toggle "use last known position" when gps signal is lost and "increase detection sensitivity when GPS is lost" so when gps signal is lost any change of LAC or SC is warning for possible catcher. This option may trigger many false alarms if it will be used in U-Bahn.
Btw with time preferences will be crowded with different togles,  so think about putting different toggles in "Advanced detection setting" menu under detection preferences?
Lock to LTE sound like a good idea, but what will hapen if you try to lock to lte where is no lte coverage? How about to check for LTE coverage before locking and if there is no LTE coverage lock to 2G?

**Comment by agilob** [comment]
Yes @SecUpwN I did understand what you mean. I just don't think it should turn itself on automatically, it causes huge battery drain.

**Comment by SecUpwN** [comment]
> I just don't think it should turn itself on automatically, it causes huge battery drain.

Me neither. But it should *remember* the its previous setting when the app gets restarted!

**Comment by agilob** [comment]
Why should?

User **should** decide when to start draining battery and getting GPS-fix.

**Comment by SecUpwN** [comment]
>User **should** decide when to start draining battery and getting GPS-fix.

Hmm.. you're somehow right as well. My initial thought was that a user could enable protections, accidentally quit our app and re-enter it without knowing that nothing is being analyzed, @larsgrefer, what do you think, would you consider this an Issue or do you have the same mindset like @agilob?

**Comment by agilob** [comment]
I don't want application to decide when it accesses my device logs and starts getting GPS-fix because it was doing last time when I used it.
